py/desispec/scripts/stdstars.py

```diff
@@ -17,200 +17,232 @@
+        for cam in frames :
+            minwave=min(minwave,np.min(frames[cam].wave))
+            maxwave=max(maxwave,np.max(frames[cam].wave))
+            mindwave=min(mindwave,np.min(np.gradient(frames[cam].wave)))
+        z_max = 0.01 # that's a huge velocity of 3000 km/s
+        minwave/=(1+z_max)
+        maxwave*=(1+z_max)
+        dwave=mindwave/2. # that's good enough
+        resampled_stdwave=minwave+dwave*np.arange(int((maxwave-minwave)/dwave))
+        log.info("first resampling of the standard star models (to go faster later) ...")
+        resampled_stdflux=np.zeros((stdflux.shape[0],resampled_stdwave.size))
+        for i in range(stdflux.shape[0]) :
+            resampled_stdflux[i]=resample_flux(resampled_stdwave,stdwave,stdflux[i])
+
+    # LOOP ON STARS TO FIND BEST MODEL
+    ############################################
+    bestModelIndex=np.arange(nstars)
+    templateID=np.arange(nstars)
+    chi2dof=np.zeros((nstars))
+    redshift=np.zeros((nstars))
     normflux=[]
-    bestModelIndex=np.arange(len(stars))
-    templateID=np.arange(len(stars))
-    chi2dof=np.zeros(len(stars))
-
-    for k,j in enumerate(stars):
-        log.info("checking best model for star {0}".format(j[0]))
-
-        starindex=j[0]
-        mags=j[2]
-        filters=fibers["FILTER"][k]
-        rflux=stars[k][1]["r"][0]
-        bflux=stars[k][1]["b"][0]
-        zflux=stars[k][1]["z"][0]
-        flux={"b":bflux,"r":rflux,"z":zflux}
-
-        #print ivars
-        rivar=ivars[k][1]["r"][0]
-        bivar=ivars[k][1]["b"][0]
-        zivar=ivars[k][1]["z"][0]
-        ivar={"b":bivar,"r":rivar,"z":zivar}
-
-        resol_star={"r":frameResolution["r"][j[0]],"b":frameResolution["b"][j[0]],"z":frameResolution["z"][j[0]]}
-
-        # Now find the best Model
-
-        bestModelIndex[k],bestmodelWave,bestModelFlux,chi2dof[k]=match_templates(frameWave,flux,ivar,resol_star,stdwave,stdflux)
-
-        log.info('Star Fiber: {0}; Best Model Fiber: {1}; TemplateID: {2}; Chisq/dof: {3}'.format(j[0],bestModelIndex[k],templateid[bestModelIndex[k]],chi2dof[k]))
+    
+    for star in range(nstars) :
+        
+        log.info("finding best model for observed star #%d"%star)
+        
+        # np.array of wave,flux,ivar,resol
+        wave = {}
+        flux = {}
+        ivar = {}
+        resolution_data = {}
+        for camera in frames :
+            band=camera[0]
+            wave[band]=frames[camera].wave
+            flux[band]=frames[camera].flux[star]
+            ivar[band]=frames[camera].ivar[star]
+            resolution_data[band]=frames[camera].resolution_data[star]
+        
+        
+        bestModelIndex[star],redshift[star],chi2dof[star],z=match_templates(wave,flux,ivar,resolution_data,stdwave,stdflux, teff, logg, feh,ncpu=args.ncpu)
+        
+        log.info('Star Fiber: {0}; TemplateID: {1}; Redshift: {2}; Chisq/dof: {3}'.format(starfibers[star],bestModelIndex[star],redshift[star],chi2dof[star]))
```

Comment:
Here, there are now two different variables. Are redshift[star] and z different?